var/spack/repos/builtin/packages/strumpack/package.py

```diff
@@ -229,51 +213,43 @@ def _test_example(self, test_prog, test_dir, test_cmd, test_args):
                 "target_link_libraries({0} ".format(test_prog) + "PRIVATE STRUMPACK::strumpack)\n"
             )
 
-        # TODO: Remove/replace once self.spec['cmake'] is available here
-        cmake_bin = self.cmake_bin(set=False)
+        with working_dir(test_dir):
+            opts = self.builder.std_cmake_args
+            opts += self.cmake_args()
+            opts += ["."]
+            cmake = self.spec["cmake"].command
+            cmake(*opts)
 
-        opts = self.std_cmake_args
-        opts += self.cmake_args()
-        opts += ["."]
+            make = which("make")
+            make(test_prog)
 
-        self.run_test(
-            cmake_bin,
-            opts,
-            [],
-            installed=False,
-            purpose="test: generating makefile",
-            work_dir=test_dir,
-        )
-        self.run_test(
-            "make", test_prog, purpose="test: building {0}".format(test_prog), work_dir=test_dir
-        )
-        with set_env(OMP_NUM_THREADS="1"):
-            self.run_test(
-                test_cmd,
-                test_args,
-                installed=False,
-                purpose="test: running {0}".format(test_prog),
-                skip_missing=False,
-                work_dir=test_dir,
-            )
+            with set_env(OMP_NUM_THREADS="1"):
+                exe = which(test_cmd)
+                exe(*test_args)
 
-    def test(self):
-        """Run the stand-alone tests for the installed software."""
-        test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
+    def test_sparse_seq(self):
+        """Run sequential test_sparse"""
+        if "+mpi" in self.spec:
+            raise SkipTest("Package must be installed with '~mpi'")
         test_exe = "test_sparse_seq"
-        test_exe_mpi = "test_sparse_mpi"
         exe_arg = [join_path("..", self.test_data_dir, "pde900.mtx")]
-        if "+mpi" in self.spec:
-            test_args = ["-n", "1", test_exe_mpi]
-            test_args.extend(exe_arg)
-            mpiexe_list = ["srun", "mpirun", "mpiexec"]
-            for mpiexe in mpiexe_list:
-                if which(mpiexe) is not None:
-                    self._test_example(test_exe_mpi, test_dir, mpiexe, test_args)
-                    break
-        else:
-            self._test_example(test_exe, test_dir, test_exe, exe_arg)
+        self._test_example(test_exe, test_exe, exe_arg)
 
+    def test_sparse_mpi(self):
+        """Run parallel test_sparse"""
+        if "+mpi" not in self.spec:
+            raise SkipTest("Package must be installed with '+mpi'")
+        test_exe_mpi = "test_sparse_mpi"
+        test_args = ["-n", "1", test_exe_mpi, join_path("..", self.test_data_dir, "pde900.mtx")]
+        mpiexe_list = ["srun", "mpirun", "mpiexec"]
+        for exe in mpiexe_list:
+            try:
+                self._test_example(test_exe_mpi, exe, test_args)
+                return
+            except:
+                continue
```

Comment:
The exception needs to be caught to pass the checks.
```suggestion
            except Exception as exc:
                print(f"Skipping {exe}: {str(exc)}")
```